### PR TITLE
Migration of demo project to o3de-extras

### PR DESCRIPTION
**Pull request by michalpelka**
This PR is a migration of the main branch of the [repository](https://github.com/RobotecAI/o3de-demo-project).
It contains ROS2 navigation demo utilizing this [sample scene](https://github.com/o3de/loft-arch-vis-sample).

### CODE REVIEW

File: Projects/ROS2NavigationExample/Assets/Importer/Examples/turtlebot4/irobot_create_description/LICENSE [hunk 1]
```diff
@@ -0,0 +1,10 @@
...
+2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.
+3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote products derived from this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
```
Comment (lumberyard-employee-dm) [inline Projects/ROS2NavigationExample/Assets/Importer/Examples/turtlebot4/irobot_create_description/LICENSE:10]: So what is the actually category of license for the turtlebot4?

File: Projects/ROS2NavigationExample/Registry/bootstrap.setreg [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+    "Amazon": {
+        "AzCore": {
+            "Bootstrap": {
+		"project_path": ".."
+            }
+        }
+    }
+}
+
```
Comment (lumberyard-employee-dm) [inline Projects/ROS2NavigationExample/Registry/bootstrap.setreg:10]: This file should be deleted.
Setting the `project_path` should not be done in the shared settings

**Review by lemonade-dm — CHANGES_REQUESTED** [review]

**Comment by spham-amzn** [comment]
Can you close this PR in favor of https://github.com/o3de/RobotVacuumSample/pull/1 ?

**Comment by michalpelka** [comment]
Closing in favor of other [pr](https://github.com/o3de/RobotVacuumSample/pull/1).

**Comment by michalpelka** [comment]
Closing in favor of https://github.com/o3de/RobotVacuumSample/pull/1.

**Pull request closed without merging**